components/feature/pwa/src/main/java/mozilla/components/feature/pwa/feature/WebAppToolbarFeature.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa.feature
+
+import android.net.Uri
+import android.view.View
+import androidx.core.net.toUri
+import androidx.core.view.isGone
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+
+/**
+ * Initializes and resets the Toolbar for a Custom Tab based on the CustomTabConfig.
+ */
+class WebAppToolbarFeature(
+    private val sessionManager: SessionManager,
+    private val toolbar: View,
+    private val sessionId: String,
+    private val manifest: WebAppManifest?
+) : Session.Observer, LifecycleAwareFeature {
+
+    init {
+        toolbar.isGone = true
+    }
+
+    override fun onUrlChanged(session: Session, url: String) {
+        toolbar.isGone = isInScope(url.toUri())
+    }
```

Comment:
This setup seems to be fragile to me. I wouldn't want this to be an actual full-featured browser fragment where we just hide the toolbar and then show it again. I think we should *actually* switch to a custom tab.

A browser may have more UI that the component does not know about and the component can't control; e.g. the little drawer above the toolbar in Fenix.

You also may want the freedom to configure Custom Tabs / PWAs differently and *not* use the same fragment configuration for both.